### ISSUE TITLE
support for MacOS arm64

### ISSUE THREAD

**Issue by jianshu93** [RasmussenLab/vamb#130]
Hello Vamb team,

With the newest version of PyTorch v1.12.1, it supports apple M1 Pro/Max series GPU, which means we can run vamb with GPU support on M1 devices. However the to_device("cuda") option does not work for MacOS while the default _coreML is used, which is apple version and will be installed by default on the mentioned devices. Since vamb is so fast with GPU, I think this could be useful for a bunch of users who wants to do some testing on their laptops.

Thanks,

Jianshu

**Comment by jakobnissen** [comment]
Good call. I'll look at it next week.

**Comment by simonrasmu** [comment]
Quick update on this as well. Tried to convert GPU code to MacOS, primarily by switching `.cuda()` to `.to(_torch.device("mps"))`. Code runs, but during training loss explodes.

**Comment by jianshu93** [comment]
It seems there are so many problems with it, especially with respect to MPS: https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+arm

Maybe we need to wait for another several month before they fix all the MPS problems.

Many thanks!

Jianshu

**Comment by jianshu93** [comment]
Hello VAMB team,

The PyTorch 1.13.1 was released this week with more support for MacOS, maybe we should restart this issue and check whether this version works for M1 GPUs. 

Thanks,

Jianshu

**Comment by jakobnissen** [comment]
This is done in the latest release, 3.0.9, and on the v4 branch.
